doc/dev/docstring_typehint.md

```diff
@@ -0,0 +1,185 @@
+        a container URI (without SAS) is accepted only when the container is public.
+        Models are trained using documents that are of the following content type - 'application/pdf',
+        'image/jpeg', 'image/png', 'image/tiff', or 'image/bmp'. Other types of content in the container is ignored.
+
+        :param str training_files_url: An Azure Storage blob container's SAS URI. A container URI (without SAS)
+            can be used if the container is public. For more information on setting up a training data set, see:
+            https://docs.microsoft.com/azure/cognitive-services/form-recognizer/build-training-data-set
+        :param bool use_training_labels: Whether to train with labels or not. Corresponding labeled files must
+            exist in the blob container if set to `True`.
+        :keyword str prefix: A case-sensitive prefix string to filter documents in the source path for
+            training. For example, when using a Azure storage blob URI, use the prefix to restrict sub
+            folders for training.
+        :keyword bool include_subfolders: A flag to indicate if subfolders within the set of prefix folders
+            will also need to be included when searching for content to be preprocessed. Not supported if
+            training with labels.
+        :keyword str model_name: An optional, user-defined name to associate with your model.
+        :keyword str continuation_token: A continuation token to restart a poller from a saved state.
+        :return: An instance of an LROPoller. Call `result()` on the poller
+            object to return a :class:`~azure.ai.formrecognizer.CustomFormModel`.
+        :rtype: ~azure.core.polling.LROPoller[~azure.ai.formrecognizer.CustomFormModel]
+        :raises ~azure.core.exceptions.HttpResponseError:
+            Note that if the training fails, the exception is raised, but a model with an
+            "invalid" status is still created. You can delete this model by calling :func:`~delete_model()`
+
+        .. versionadded:: v2.1
+            The *model_name* keyword argument
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sample_train_model_without_labels.py
+                :start-after: [START training]
+                :end-before: [END training]
+                :language: python
+                :dedent: 8
+                :caption: Training a model (without labels) with your custom forms.
+        """
+        ...
+```
+
+The first portion of this docstring is a general description of what the method does. Following the general description of the method is a **required new line** and then documentation for each of the parameters, optional keyword arguments, returned objects, and potentially raised errors.
+
+Required parameters can be documented in one-line or two-lines. Both options can be used in a single docstring for documenting different parameters without issue.
```

Comment:
Not "required", but positional:

```
def (a,b,c=None)
```
They are all positional and documented with `param`, but they are not all required.